Common/Kernel/CommonValidations.cs

```diff
@@ -13,5 +13,18 @@ public static void ValidateId(int id)
                 throw new Exception(Errors.IdCanNotBeLessThanZero);
             }
         }
+        
+        public static void ValidateEmail(string email)
+        {
+
+            var addr = new MailAddress(email); 
+            bool emailIsValid = addr.Address == email;
+            
+            if (emailIsValid == false)
+            {
+                throw new Exception(Errors.EmailCanNotBeNullEmptyOrNotEmailFormat);

```

Comment:
May be we should use another type of Exception? For example, a custom one.

Reply:
It is a custom exception that I created

Reply:
I mean a custom exception as a class, inheriting `Exception`, not a custom error message

Reply:
It is not a custom exception, Nikita is right.